deduplicate installation instructions (#9507)
NixOS/nix: doc/manual/src/installation/installing-binary.md

  - macOS
- [Single-user](#single-user-installation):
  - Linux without systemd
  - Linux with SELinux

We recommend the multi-user installation if it supports your platform and you can authenticate with `sudo`.

The installer can configured with various command line arguments and environment variables.
To show available command line flags:

```console
$ curl -L https://nixos.org/nix/install | sh -s -- --help
```

To check what it does and how it can be customised further, [download and edit the second-stage installation script](#installing-from-a-binary-tarball).

# Installing a pinned Nix version from a URL

Version-specific installation URLs for all Nix versions since 1.11.16 can be found at [releases.nixos.org](https://releases.nixos.org/?prefix=nix/).
The directory for each version contains the corresponding SHA-256 hash.

These installation scripts can be used the same as usual:

```console
$ VERSION=2.19.2 curl -L https://releases.nixos.org/nix/nix-$VERSION/install | sh
```

# Multi User Installation

The multi-user Nix installation creates system users and a system service for the Nix daemon.

Supported systems:

- Linux running systemd, with SELinux disabled
- macOS

> You can explicitly instruct the installer to perform a multi-user installation on your system:

Author: ```suggestion
To explicitly instruct the installer to perform a multi-user installation on your system:
```

# Installing a Binary Distribution

To install the latest version Nix, run the following command:

```console
$ curl -L https://nixos.org/nix/install | sh
```

This will run the installer interactively, that will explain what it is doing.
It will perform the default type of installation for your platform:

- [Multi-user](#multi-user-installation):
  - Linux with systemd and without SELinux
  - macOS
- [Single-user](#single-user-installation):
  - Linux without systemd
  - Linux with SELinux

We recommend the multi-user installation if it supports your platform and you can authenticate with `sudo`.

> The installer can configured with various command line arguments and environment variables.

Author: Why *also*? It's the primary way of configuring it.

To show available command line flags:

```console
$ curl -L https://nixos.org/nix/install | sh -s -- --help
```

To check what it does and how it can be customised further, [download and edit the second-stage installation script](#installing-from-a-binary-tarball).

# Installing a pinned Nix version from a URL

Version-specific installation URLs for all Nix versions since 1.11.16 can be found at [releases.nixos.org](https://releases.nixos.org/?prefix=nix/).
The corresponding SHA-256 hash can be found in the directory for the given version.

> These installation scripts can be used the same as usual:

Reviewer: ```suggestion
You can run these installation scripts the same way:
```

```console

> $ VERSION=2.19.2 curl -L https://releases.nixos.org/nix/nix-$VERSION/install | sh

Reviewer: At least in POSIXy shells, I think these have to get split up across two lines if it's going to work. (Or keep telling the user to manually write it in as the previous text did).

Parameter assignments prefixed to an invocation will be exported to the environment of the running command right before it's executed, but  parameter expansion happens very early on in the process of evaluating input from the user:

```
$ bash -c 'VERSION=2.19.2 echo curl -L https://releases.nixos.org/nix/nix-$VERSION/install'
curl -L https://releases.nixos.org/nix/nix-/install

$ zsh -c 'VERSION=2.19.2 echo curl -L https://releases.nixos.org/nix/nix-$VERSION/install'
curl -L https://releases.nixos.org/nix/nix-/install

$ fish -c 'VERSION=2.19.2 echo curl -L https://releases.nixos.org/nix/nix-$VERSION/install'
curl -L https://releases.nixos.org/nix/nix-2.19.2/install

$ dash -c 'VERSION=2.19.2 echo curl -L https://releases.nixos.org/nix/nix-$VERSION/install'
curl -L https://releases.nixos.org/nix/nix-/install
```

Author: Ah yes

Author: ```suggestion
$ export VERSION=2.19.2 
$ curl -L https://releases.nixos.org/nix/nix-$VERSION/install | sh
```

```

# Multi User Installation

The multi-user Nix installation creates system users, and a system service for the Nix daemon.

Supported systems:

- Linux running systemd, with SELinux disabled
- macOS

You can explicitly instruct the installer to perform a multi-user installation on your system:

```console
$ curl -L https://nixos.org/nix/install | sh -s -- --daemon
```

You can run this under your usual user account or `root`.
The script will invoke `sudo` as needed.

# Single User Installation

To explicitly select a single-user installation on your system:

```console
$ curl -L https://nixos.org/nix/install | sh -s -- --no-daemon
```

> This will perform a single-user installation of Nix, meaning that `/nix` is owned by the invoking user.

Reviewer: ```suggestion
In a single-user installation, `/nix` is owned by the invoking user.
```

> You can run this under your usual user account or `root`.

Reviewer: Is this root bit true? I guess I've always taken the source at its word, here:

https://github.com/NixOS/nix/blob/188c803ddb5e63b243ddb84eba9b70e45475b7ea/scripts/install-nix-from-closure.sh#L135-L137

It looks like the language was changed from "not recommended" to "not supported" way back in 5d064e2.

Reviewer: Also: I'm reading the previous sentence as trying to draw a distinction between multi- and single-user installs--but the assertion that we can invoke this as `root` seems to undercut or muddle that distinction.

Author: ```suggestion
```

I was in fact also confused, but didn't bother to check. It was enough of a change as it is.

The script will invoke `sudo` to create `/nix` if it doesn’t already exist.
If you don’t have `sudo`, manually create `/nix` as `root`:

```console
$ su root
# mkdir /nix
# chown alice /nix
```

# Installing from a binary tarball

You can also download a binary tarball that contains Nix and all its dependencies:
- Choose a [version](https://releases.nixos.org/?prefix=nix/) and [system type](../contributing/hacking.md#platforms)
- Download and unpack the tarball
- Run the installer

> **Example**
>
> ```console
> $ pushd $(mktemp -d)
> $ export VERSION=2.19.2
> $ export SYSTEM=x86_64-linux
> $ curl -LO https://releases.nixos.org/nix/nix-$VERSION/nix-$VERSION-$SYSTEM.tar.xz
> $ tar xfj nix-$VERSION-$SYSTEM.tar.xz
> $ cd nix-$VERSION-$SYSTEM-darwin
> $ ./install
> $ popd
> ```

> The installer can be customised using environment varibales set in the file named `install-multi-user`.

Author: Your suggestion merely switches to "you can" active voice, it doesn't change the framing. Maybe the wording is potentially misleading, because those env vars *are* set in the file, but it may be read as "which *you* set (wherever)". 

```suggestion
The installer can be customised using environment variables declared in the file named `install-multi-user`.
```
